docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,275 @@
+- we'd need to relax the requirement of mTLS (it's only needed for services) to run egress
+- there is additional network hop
+- there is a load balancing issue with TCP traffic using zone egress when we make a connection and change the load
+  balancing algorithm we have to wait for the connection to be re-created for the reconfiguration to take place
+- with the introduction of MeshPassthrough the barrier for external traffic is lower and some cases can be covered by that
+
+Conclusion: **it seems like we're not cutting any serious use case**.
+
+### Running traffic through sidecar only
+
+- all the problems of running through egress are not applicable (operational cost, extra networking hop, scalability etc.)
+- no single exit point for all the external traffic
+
+### Running traffic through sidecar and egress
+
+- we can mix and match the pros/cons of both approaches
+- it's harder to implement / explain how it's working
+
+### Configurable where it makes sense on both sidecar and egress, predefined where only one makes sense
+
+#### Table summary
+
+The table below shows a summary of policies and if it makes sense to configure them on:
+- sidecar
+- egress
+- on both
+
+and if it makes sense for the user to pick where.
+
+| Policy                    | On sidecar    | On egress              | On both | User configurable |
+|---------------------------|---------------|------------------------|---------|-------------------|
+| MeshAccessLog             | Yes (in `to`) | Yes (both `to`/`from`) | Yes     | Yes               |
+| MeshCircuitBreaker        | Yes           | Maybe                  | No      | No                |
+| MeshFaultInjection        | Maybe         | Yes                    | No      | No                |
+| MeshHealthCheck           | Yes           | Maybe                  | No      | No                |
+| MeshMetric                | Yes           | Yes                    | Yes     | Maybe             |
+| MeshProxyPatch            | Yes           | Yes                    | Yes     | Yes               |
+| MeshRateLimit             | Probably not  | Yes                    | No      | No                |
+| MeshRetry                 | Yes           | Maybe                  | No      | Maybe             |
+| MeshTimeout               | Yes           | Maybe                  | No      | No                |
+| MeshTrace                 | Yes           | Yes                    | Yes     | Maybe             |
+| MeshTrafficPermission     | Maybe         | Yes                    | Maybe   | Maybe             |
+| MeshLoadBalancingStrategy | Yes           | Yes                    | No      | Maybe             |
+| MeshTCPRoute              | Yes           | Yes                    | No      | No                |
+| MeshHTTPRoute             | Yes           | Yes                    | No      | No                |
```

Comment:
how would routes work on egress only? What's the use case?

Reply:
can you elaborate on the problems there?

Reply:
1) How do you target egress and specific MeshExternalService
2) Can you redirect just to other MeshExternalService or to MeshServices as well?

Reply:
> 1. How do you target egress and specific MeshExternalService

it's the same question as in https://github.com/kumahq/kuma/pull/10921/files#diff-be8076fceda3a49a1fa26af0d8cab5bfb664c6e87fd1a0dccbb4fac043629484R270 section. It's either: `kind: Mesh`, `proxyType` + `sectionName` or use `MeshExternalService` as a top level target + `proxyType: egress` + `name`.

> 2. Can you redirect just to other MeshExternalService or to MeshServices as well?

I don't think you should be able to redirect to other Kinds.

Reply:
it's `to` policy so it's on the sidecar